lib/travis/guest-api/app/endpoint/steps.rb

```diff
@@ -8,44 +8,59 @@ class Steps < Travis::GuestApi::App::Endpoint
     end
 
     post '/steps' do
-      steps = env['rack.parser.result']
-      steps = [ params ] unless (Array === env['rack.parser.result'])
+      begin
+        steps = env['rack.parser.result']
+        steps = [ params ] unless (Array === env['rack.parser.result'])
+
+        steps.map! do |step|
+
+          raise Travis::GuestAPI::Cache::AddStepException,
+            'Keys name, classname are mandatory!' unless !step.nil? &&
```

Comment:
Just aesthetical suggestion:
"Mandatory parameters 'name' and 'classname' need to be specified'"